### DISCUSSION TITLE
[feat] Possibility to notify about PR/MR creation via webhook (i.e. slack)

### DISCUSSION THREAD

**Discussion by grzesuav** [renovatebot/renovate#22146] [Suggest an Idea]
**What would you like Renovate to be able to do?**
I would like to have possibility to configure renovate to notify on slack when PR/MR is created.

**Describe the solution you'd like**
Possibility to configure :
* webhook url
* secret in configuration file // env variable ? - not sure how it could be handled
* message template

**Describe alternatives you've considered**
Notifications via email/reviewer - slack would fit better in my workflow.

**Comment by ChristianMurphy** [comment]
@grzesuav would linking to documentation how to integrate GitHub/GitLab/BitBucket with Slack address your use case?
https://slack.github.com
https://confluence.atlassian.com/bitbucket/bitbucket-cloud-for-slack-945096776.html
https://docs.gitlab.com/ee/user/project/integrations/slack.html

Or is there some renovate specific information that would be posted to Slack? (if so, what would that info be?)

**Comment by grzesuav** [comment]
@ChristianMurphy I know about gitlab/slack integration, however :
* once enabled, it notifies about **every** MR even, like creation, merge, closed etc - this is too much spam
* I would like to be notified only about renovate MR's, there is no possibility to configure it with existing apis (to limit MR events to specific author etc)

**Comment by rarkins** [comment]
I think that providing a flexible way to configure webhooks for select events is a nice feature idea, although it’s a feature that could get quite complex quickly, Eg:
- which events can trigger a webhooks
- multiple webhook receivers
- auth for webhooks

If there’s any other service with a similar approach to webhook configuration then it would certainly help to lock down requirements here. 

- Multiple events can trigger webhooks, Eg PR creation, PR updates, automerges, vulnerability PR creation, etc
- each event can have more than one webhook receiver
- presumably need configurable get vs post
- need a configurable JSON body (eg field names, values)

**Comment by gekkedev** [comment]
I'd like to throw in another approach which could be either a short-term goal or also an addition to keep Renovate unopinionated:   Void callbacks to let people chain anything they want. There would be a new setting in the config file, e.g. 
```json
"callbackScriptFile": "./myCallback.js",
```
 or to make it even more generic:
```json
"postPrCreationCallbackCmd": "node-ts ./myPostPrCallback.ts"
```
 (added `prCreation` segment as there might be a demand for event differentiation whilst an interfaced JS file can just have a default export that take this as string argument to decide by itself what to do)
What if, let's say a certain service's webhook, is not covered by the requirements? Then this kind of hook would be easier to implement than a full API integration.

**Comment by shlomioz** [comment]
Hey guys, there is any news about this feature?

**Comment by dr460neye** [comment]
I would not recommend to use the gitlab or other git features. Reason here are data privacy conflicts. When the bot is hosted by a different provider you will face problems in Europe due to employee rights, as you loose control company wide.

**Comment by fabb** [comment]
We are self-hosting both
